### PR TITLE
Chavanr/e2e cohort review participant detail

### PR DESCRIPTION
**Pull request by chavanr2019**
updated the cohort-review test case with the following steps:
1. add Annotations fields
2. edit Annotation field name
3. delete annotation field name
---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/tests/cohorts/cohorts-review.spec.ts [hunk 1]
```diff
@@ -119,11 +120,38 @@ describe('Cohort review tests', () => {
...
+
+    await participantDetailPage.goToThePriorParticipant();
+    await participantDetailPage.waitForLoad();
+
+    // verify that the prior participant is displaying the same annotation field name
+    expect(annotationNewTextBoxName).toEqual(newAnnotationRename);
+
+    // click the annotations edit  button to delete the annotation textbox field
+    await sidebarContent.getAnnotationsEdit().then(btn => btn.click());
+    await editDeleteAnnotationsFieldModal.deleteAnnotationsName();
```
Comment (aweng98) [inline e2e/tests/cohorts/cohorts-review.spec.ts:151]: do you think delete textbox needs a verification?

File: e2e/app/component/edit-delete-annotations-modal.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+   /**
+    * @param {string} newName New name.
+    */
+  
+  // rename annotations field  name
+  async clickRenameAnnotationsName(newName?: string): Promise<void> {
+     await this.clickButton(LinkText.Rename);
+      
+     const selector = `${this.getXpath()}//*[contains(text(), "Edit")]/following::div[3]/input`;
```
Comment (aweng98) [inline e2e/app/component/edit-delete-annotations-modal.ts:39]: this xpath is prone to breaking. does `Textbox` class not working for this?

File: e2e/app/component/edit-delete-annotations-modal.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const modalTitle = 'Edit or Delete Review-Wide Annotation Fields';
+
+
+export default class EditDeleteAnnotationsModal extends Modal {
+
+  constructor(page: Page, xpath?: string) {
+    super(page, xpath);
+  }
+
+  async waitForLoad(): Promise<this> {
```
Comment (aweng98) [inline e2e/app/component/edit-delete-annotations-modal.ts:18]: are you overriding `waitForLoad` function in `Modal` class intentionally?

File: e2e/tests/cohorts/cohorts-review.spec.ts (hunk 1 above)
Comment (chavanr2019) [reply e2e/tests/cohorts/cohorts-review.spec.ts:151]: Agree! we need to have a verification step for delete textbox. Will add it!

File: e2e/app/component/edit-delete-annotations-modal.ts (hunk 2 above)
Comment (chavanr2019) [reply e2e/app/component/edit-delete-annotations-modal.ts:39]: Right! `Textbox `class not working here. In this case, when the rename button is clicked 
1. the cursor shows up in the textbox input field.
2. the existing text needs to be cleared.
3. then the newName is typed.

File: e2e/app/component/edit-delete-annotations-modal.ts (hunk 3 above)
Comment (chavanr2019) [reply e2e/app/component/edit-delete-annotations-modal.ts:18]: @aweng98 frankly, I did not give much thought to it! Here I've reused the `new-notebook-modal.ts` file.

**Review by aweng98 — APPROVED** [review]
Looks good. Feel free to merge anytime after update.

File: e2e/tests/cohorts/cohorts-review.spec.ts (hunk 1 above)
Comment (chavanr2019) [reply e2e/tests/cohorts/cohorts-review.spec.ts:151]: @aweng98 I tried adding a verification step for the textbox deletion but did not succeed.

File: e2e/app/component/sidebar-content.ts [hunk 4]
```diff
@@ -61,10 +64,46 @@ export default class SidebarContent extends  HelpSidebar {
...
+     async findFieldName(annotationFieldName: string): Promise<string | null> {
+        try {
+          this.getFieldNameXpath(`${annotationFieldName}`);
+        } catch (err) {
+          // no field.
+          return null;
+        }
+    }
+  
 }
```
Comment (aweng98) [inline e2e/app/component/sidebar-content.ts:109]: needs a newline break

File: e2e/app/component/sidebar-content.ts [hunk 5]
```diff
@@ -61,10 +64,46 @@ export default class SidebarContent extends  HelpSidebar {
...
+    }
+
+     // click on the Annotations EDIT button to edit the annotations field name
+     async getAnnotationsEdit(): Promise<Button>{
+      return Button.findByName(this.page, {name: LinkText.Edit});
+    }
+
+     // text area
+     async getAnnotationsTextArea(): Promise<Textarea> {
+      const selector = `${this.getXpath()}//*[contains(normalize-space(text()), "Annotations")]/following::textarea`;
```
Comment (aweng98) [inline e2e/app/component/sidebar-content.ts:94]: single quote. not backtick.

File: e2e/app/text-labels.ts [hunk 6]
```diff
@@ -79,6 +82,8 @@ export enum LinkText {
    StayHere = 'Stay Here',
    Submit = 'Submit',
    Update = 'Update',
+   Yes = 'Yes',
+
```
Comment (aweng98) [inline e2e/app/text-labels.ts:86]: remove empty space

File: e2e/app/component/sidebar-content.ts [hunk 7]
```diff
@@ -61,10 +64,46 @@ export default class SidebarContent extends  HelpSidebar {
     }
    
     // get the annotations name displaying on the sidebar panel
-    async getAnnotationsName(): Promise<string> {
-      const xpath = '//*[contains(normalize-space(text()), "Annotations")]/following::div[4]';
-      const element = BaseElement.asBaseElement(this.page, await this.page.waitForXPath(xpath, {visible: true}));
-      return element.getTextContent();
+    async getAnnotationsName(annotationFieldName:string): Promise<string> {
+    const selector = this.getFieldNameXpath(`${annotationFieldName}`);
+    const annotationfieldLabel = await this.extractFieldNameText(`${selector}`);
```
Comment (aweng98) [inline e2e/app/component/sidebar-content.ts:69]:
```
.extractFieldNameText(`${selector}`
```
to 
`.extractFieldNameText(selector`

File: e2e/app/component/sidebar-content.ts [hunk 8]
```diff
@@ -61,10 +64,46 @@ export default class SidebarContent extends  HelpSidebar {
     }
    
     // get the annotations name displaying on the sidebar panel
-    async getAnnotationsName(): Promise<string> {
-      const xpath = '//*[contains(normalize-space(text()), "Annotations")]/following::div[4]';
-      const element = BaseElement.asBaseElement(this.page, await this.page.waitForXPath(xpath, {visible: true}));
-      return element.getTextContent();
+    async getAnnotationsName(annotationFieldName:string): Promise<string> {
```
Comment (aweng98) [inline e2e/app/component/sidebar-content.ts:67]: `annotationFieldName:string` needs a whitespace before `string`

File: e2e/app/component/sidebar-content.ts [hunk 9]
```diff
@@ -61,10 +64,46 @@ export default class SidebarContent extends  HelpSidebar {
...
+
+     // click on the Annotations EDIT button to edit the annotations field name
+     async getAnnotationsEdit(): Promise<Button>{
+      return Button.findByName(this.page, {name: LinkText.Edit});
+    }
+
+     // text area
+     async getAnnotationsTextArea(): Promise<Textarea> {
+      const selector = `${this.getXpath()}//*[contains(normalize-space(text()), "Annotations")]/following::textarea`;
+      const annotationTextArea = new Textarea(page, selector);
```
Comment (aweng98) [inline e2e/app/component/sidebar-content.ts:95]: change 
```
new Textarea(page, selector);
```
to 
`new Textarea(this.page, selector);`

Comment (chavanr2019) [reply e2e/app/component/sidebar-content.ts:95]: @aweng98 Thanks for the review... will revise the code soon!

File: e2e/app/component/sidebar-content.ts (hunk 5 above)
Comment (chavanr2019) [reply e2e/app/component/sidebar-content.ts:94]: Hi @aweng98 could you confirm about having a single quote here and not backtick?

Comment (aweng98) [reply e2e/app/component/sidebar-content.ts:94]: I'm wrong. backtick is right here. I think I didn't see `${this.getXpath()`

**Review by aweng98 — APPROVED** [review]

**Pull request merged**